File: step_definitions/Order Confirmation Email.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class OrderConfirmationSteps extends OrderConfirmationPage {

    private TestHarness testHarness = new TestHarness();

    @Given("a user completes a purchase on the retail platform")
    public void userCompletesPurchase() {
        completePurchase();
    }

    @When("the transaction is successfully processed")
    public void transactionProcessed() {
        processTransaction();
    }

    @Then("an order confirmation email should be sent to the user's registered email address")
    public void orderConfirmationEmailSent() {
        verifyEmailSent();
    }

    @Then("the email should contain details of the transaction including order number, items purchased, total amount, and billing information")
    public void verifyEmailContent() {
        verifyEmailDetails();
    }

    @Then("the user should receive the email within a reasonable time frame after the purchase completion")
    public void verifyEmailReceiptTime() {
        verifyEmailReceiptTime();
    }

    @Given("a user attempts to make a purchase on the retail platform")
    public void userAttemptsPurchase() {
        attemptPurchase();
    }

    @When("the transaction fails")
    public void transactionFails() {
        failTransaction();
    }

    @Then("no order confirmation email should be sent to the user")
    public void noEmailSent() {
        verifyNoEmailSent();
    }

    @Given("a user has completed a purchase")
    public void userHasCompletedPurchase() {
        completePurchase();
    }

    @When("the order confirmation email is received")
    public void emailReceived() {
        receiveEmail();
    }

    @Then("the email should contain the correct order number")
    public void verifyOrderNumber() {
        verifyOrderNumber();
    }

    @Then("the email should list all the items purchased")
    public void verifyItemsPurchased() {
        verifyItemsPurchased();
    }

    @Then("the email should show the correct total amount")
    public void verifyTotalAmount() {
        verifyTotalAmount();
    }

    @Then("the email should include accurate billing information")
    public void verifyBillingInformation() {
        verifyBillingInformation();
    }

    @Given("a user has completed a purchase")
    public void userCompletedPurchase() {
        completePurchase();
    }

    @Given("the user requests a resend of the order confirmation email")
    public void userRequestsResend() {
        requestEmailResend();
    }

    @When("the request is processed")
    public void processResendRequest() {
        processResendRequest();
    }

    @Then("the order confirmation email should be resent to the user's registered email address")
    public void emailResent() {
        verifyEmailResent();
    }

    @Given("a user completes a purchase")
    public void userCompletesPurchaseAgain() {
        completePurchase();
    }

    @Given("the user's registered email address is invalid")
    public void invalidEmailAddress() {
        setInvalidEmailAddress();
    }

    @When("the system attempts to send the order confirmation email")
    public void systemAttemptsEmailSend() {
        attemptEmailSend();
    }

    @Then("the system should log an error indicating the invalid email address")
    public void logInvalidEmailError() {
        logEmailError();
    }

    @Then("no order confirmation email should be sent")
    public void noEmailSentDueToInvalidAddress() {
        verifyNoEmailSent();
    }

    @Then("the user should be notified to update their email address")
    public void notifyUserToUpdateEmail() {
        notifyUserToUpdateEmail();
    }
}

File: toot/step_definitions/Order Confirmation Email.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import OrderConfirmationPage from "../pages/orderConfirmationPage";

Given("a user has completed a purchase", () => {
 OrderConfirmationPage.completePurchase();
});

When("the purchase transaction is successful", () => {
 OrderConfirmationPage.verifyTransactionSuccess();
});

Then("an order confirmation email should be sent to the user's registered email address", () => {
 OrderConfirmationPage.verifyEmailSent();
});

Then("the email should contain the order details including order ID, items purchased, total amount, and delivery address", () => {
 OrderConfirmationPage.verifyEmailContent();
});

Then("the email should be sent from the correct sender email address for order confirmations", () => {
 OrderConfirmationPage.verifySenderEmailAddress();
});

Given("a user attempts to complete a purchase", () => {
 OrderConfirmationPage.attemptPurchase();
});

When("the purchase transaction fails", () => {
 OrderConfirmationPage.verifyTransactionFailure();
});

Then("no order confirmation email should be sent to the user", () => {
 OrderConfirmationPage.verifyNoEmailSent();
});

Given("a user has completed a purchase", () => {
 OrderConfirmationPage.completePurchase();
});

Given("the initial attempt to send an order confirmation email fails", () => {
 OrderConfirmationPage.failInitialEmailAttempt();
});

When("the system retries sending the email", () => {
 OrderConfirmationPage.retryEmailSending();
});

Then("the order confirmation email should be successfully sent to the user's registered email address", () => {
 OrderConfirmationPage.verifyEmailSent();
});

Given("a user has completed a purchase", () => {
 OrderConfirmationPage.completePurchase();
});

When("an order confirmation email is sent", () => {
 OrderConfirmationPage.sendEmail();
});

Then("the system should log the status of the email sending process", () => {
 OrderConfirmationPage.logEmailStatus();
});

Then("the log should include whether the email was successfully sent or if there were any errors", () => {
 OrderConfirmationPage.verifyLogContent();
});